### DISCUSSION TITLE
[V5] Redefining Route.resource routes

### DISCUSSION THREAD

**Discussion by Spedwards** [adonisjs/core#2442] [Help]
I'd like to change up a couple of the routes that `Route.resource` provides. Is there anyway to redefine it?

**Comment by RomainLanz** [comment]
Hey @Spedwards! 👋 

You can use `.only()` and `.except()` to remove or only-add routes from `Route.resource`.

```ts
Route
  .resource('comments', 'CommentsController')
  .except(['update', 'destroy'])

// or

Route
  .resource('comments', 'CommentsController')
  .only(['index', 'create', 'store', 'show', 'edit'])
```

📚 [Filtering Resourceful Routes](https://preview.adonisjs.com/guides/http/routing#filtering-resourceful-routes)

Then, you can add your custom routes by defining them directly.

```ts
Route.get('comments/foo', 'CommentsController.foo')

Route
  .resource('comments', 'CommentsController')
  .except(['update', 'destroy'])
```

Note that I am defining the custom route before the `Route.resource`.
This is needed because the `show` route of the resource will catch anything after `comments/` (`comments/foo` will be caught by `/comments/{id}`).

**Reply by Spedwards** [reply]
That's honestly unfortunate. I know I can define single routes, however I'm going to need a lot of routes in the same pattern, which is why I wanted to extend `Route.resource`.

**Reply by RomainLanz** [reply]
What kind of extension do you need?
You can still create your own helper if you need to.

**Reply by Spedwards** [reply]
I'd want to add a couple routes to `Route.resource` in addition to changing the existing routes paths. Currently I'd just not use `Route.resource` and define all my routes manually for each controller however I'd like to use `Route.resource` if possible to avoid defining everything manually.

**Reply by thetutlage** [reply]
`Route.resource` uses a convention of Restful routes. Modifying what is generates isn't following the convention anymore. Also, if you are not following the convention, then you have to manually define all the routes and there is no shortcut to it.
